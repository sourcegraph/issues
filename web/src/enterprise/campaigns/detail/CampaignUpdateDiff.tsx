import * as React from 'react'
import * as GQL from '../../../../../shared/src/graphql/schema'
import { ThemeProps } from '../../../../../shared/src/theme'
import AlertCircleIcon from 'mdi-react/AlertCircleIcon'
import { HeroPage } from '../../../components/HeroPage'

interface Props extends ThemeProps {
    campaign: Pick<GQL.ICampaign, 'id' | 'viewerCanAdminister'> & {
        changesets: Pick<GQL.ICampaign['changesets'], 'totalCount'>
    }
    className?: string
}

/**
 * A list of a campaign's changesets changed over a new patch set
 */
export const CampaignUpdateDiff: React.FunctionComponent<Props> = ({ campaign, className }) => {
    if (!campaign.viewerCanAdminister) {
        return <HeroPage body="Updating a campaign is not permitted without campaign admin permissions." />
    }
    return (
        <div className={className}>
            <h3 className="mt-4 mb-2">Preview of changes</h3>
            <div className="alert alert-info mt-2">
                <AlertCircleIcon className="icon-inline" /> You are updating an existing campaign. By clicking 'Update',
                all above changesets that are not 'unmodified' will be updated on the codehost.
            </div>
        </div>
    )
}
