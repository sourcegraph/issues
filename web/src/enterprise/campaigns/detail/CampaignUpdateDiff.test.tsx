import React from 'react'
import { CampaignUpdateDiff } from './CampaignUpdateDiff'
import { mount } from 'enzyme'

describe('CampaignUpdateDiff', () => {
    test('renders', () => {
        expect(
            mount(
                <CampaignUpdateDiff
                    isLightTheme={true}
                    campaign={{
                        id: 'somecampaign',
                        changesets: { totalCount: 1 },
                        viewerCanAdminister: true,
                    }}
                />
            )
        ).toMatchSnapshot()
    })
})
