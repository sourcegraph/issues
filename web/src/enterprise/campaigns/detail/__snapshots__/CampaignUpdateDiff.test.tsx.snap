// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignUpdateDiff renders 1`] = `
<CampaignUpdateDiff
  campaign={
    Object {
      "changesets": Object {
        "totalCount": 1,
      },
      "id": "somecampaign",
      "viewerCanAdminister": true,
    }
  }
  isLightTheme={true}
>
  <div>
    <h3
      className="mt-4 mb-2"
    >
      Preview of changes
    </h3>
    <div
      className="alert alert-info mt-2"
    >
      <Memo(AlertCircleIcon)
        className="icon-inline"
      >
        <svg
          className="mdi-icon icon-inline"
          fill="currentColor"
          height={24}
          viewBox="0 0 24 24"
          width={24}
        >
          <path
            d="M13,13H11V7H13M13,17H11V15H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          />
        </svg>
      </Memo(AlertCircleIcon)>
       You are updating an existing campaign. By clicking 'Update', all above changesets that are not 'unmodified' will be updated on the codehost.
    </div>
  </div>
</CampaignUpdateDiff>
`;
